### PR TITLE
Create archivematica user group in rpms

### PR DESCRIPTION
**Pull request by scollazo**
Also, improve user handling.

This fixes artefactual-labs/am-packbuild#116 and
artefactual/archivematica#955

### CODE REVIEW

File: rpm/archivematica-storage-service/package.spec [hunk 1]
```diff
@@ -76,12 +76,17 @@ rm -rf %{buildroot}
...
-  useradd --uid 333 --user-group --home /var/lib/archivematica/ archivematica
+# Create archivematica user and group
+getent group archivematica >/dev/null || groupadd -f -g 333 -r archivematica
+if ! getent passwd archivematica >/dev/null ; then
+    if ! getent passwd 333 >/dev/null ; then
+      useradd -r -u 333 -g archivematica -d /var/lib/archivematica/ -s /sbin/nologin -c "Archivematica system account" archivematica
+    else
+      useradd -r -g archivematica -d /var/lib/archivematica/ -s /sbin/nologin -c "Archivematica system account" archivematica
+    fi
 fi
```
Comment (sevein) [inline rpm/archivematica-storage-service/package.spec]: The indentation in the block above is inconsistent, it mixes 2 (L83 and L85) and 4 whitespaces.

File: rpm/archivematica-storage-service/package.spec [hunk 2]
```diff
@@ -76,12 +76,17 @@ rm -rf %{buildroot}
 
 %post
 
-# Create archivematica user
-userID=`id -u archivematica`
-if [ "${userID}" != 333 ]; then
-  useradd --uid 333 --user-group --home /var/lib/archivematica/ archivematica
+# Create archivematica user and group
+getent group archivematica >/dev/null || groupadd -f -g 333 -r archivematica
```
Comment (sevein) [inline rpm/archivematica-storage-service/package.spec]: I've seen that this snippet comes from http://fedoraproject.org/wiki/Packaging%3aUsersAndGroups#Soft_static_allocation but I don't understand why we need `getent group archivematica >/dev/null || ` considering that `groupadd -f` is going to exit successfully regardless. Can you think of a reason? I've read their rationale but still it doesn't make sense in this case.

```
$ docker run -it centos:7 bash
% groupadd -g 12345 -r archivematica
% groupadd -f -g 333 -r archivematica
% getent group 333
% getent group 12345
archivematica:x:12345:
```

File: rpm/archivematica-storage-service/package.spec (hunk 1 above)
Comment (sevein) [reply rpm/archivematica-storage-service/package.spec]: Should we use variables to make this a bit clearer?
HOMEDIR, USERNAME, GROUPNAME, ALLOCATED_UID and ALLOCATED_GID?

**Comment by scollazo** [comment]
As explained in the Fedora site, if the system administrator wants to create the user in advance, and assign a different UID/GID , getent will fail and the user won't be created

In the original script, there is an "exit 0", so in case of problems creating the user, it ends succesfully .I didn't use it, as we are running the user creation in the postscript, not in the pre section as fedora does, and I prefer to be aware of posible problems.

**Comment by sevein** [comment]
I still think that the bit I mentioned in the snippet you copied from them may be redundant but I guess it's not a big deal.

**Pull request merged**